src/acceptance-test/java/org/zalando/nakadi/webservice/hila/SubscriptionLoadAT.java

```diff
@@ -0,0 +1,266 @@
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.apache.http.HttpStatus;
+import org.junit.Assert;
+import org.junit.Test;
+import org.zalando.nakadi.config.JsonConfig;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.Subscription;
+import org.zalando.nakadi.exceptions.Try;
+import org.zalando.nakadi.view.SubscriptionCursor;
+import org.zalando.nakadi.webservice.BaseAT;
+import org.zalando.nakadi.webservice.utils.NakadiTestUtils;
+import org.zalando.nakadi.webservice.utils.TestStreamingClient;
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.http.ContentType.JSON;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.zalando.nakadi.utils.TestUtils.waitFor;
+import static org.zalando.nakadi.webservice.hila.SubscriptionAT.SUBSCRIPTIONS_URL;
+
+public class SubscriptionLoadAT extends BaseAT {
+
+    private static final int EVENT_TYPE_COUNT = 2;
+    private static final int PARTITION_PER_EVENT_TYPE = 8;
+    private static final int MESSAGES_PER_EVENT_TYPE = PARTITION_PER_EVENT_TYPE * 10;
+    private static final ObjectMapper MAPPER = (new JsonConfig()).jacksonObjectMapper();
+
+    private static class TimeLogger {
+        private long start = System.currentTimeMillis();
+
+        public void mark(final String message) {
+            final long now = System.currentTimeMillis();
+            System.out.println(message + " took " + (now - start));
+            start = now;
+        }
+    }
+
+    /**
+     * Test does:
+     * - creates 2 event types with 8 partitions each
+     * - posts 10 events to each partition
+     * - creates subscription for all event types
+     * - consumes all events with 7 consumers at the same time
+     * - checks that everything was consumed
+     * - measures time of consumption
+     * - checks if every consumer was used at least once
+     * - checks that data was delivered only once
+     */
+    @Test(timeout = 30000)
+    public void testSubscriptionLoad() throws IOException, InterruptedException {
+        // create event types
+        final TimeLogger timeLogger = new TimeLogger();
+        final List<EventType> eventTypeList = createEventTypes(EVENT_TYPE_COUNT, PARTITION_PER_EVENT_TYPE);
+        timeLogger.mark("Create event types");
+
+        // publish events to each event type
+        eventTypeList.parallelStream().forEach(et -> sendEvents(et, MESSAGES_PER_EVENT_TYPE, PARTITION_PER_EVENT_TYPE));
+        timeLogger.mark("Publishing " + (MESSAGES_PER_EVENT_TYPE * EVENT_TYPE_COUNT) + " messages");
+
+        // create subscription
+        final Subscription subscription = createSubscription(eventTypeList);
+        timeLogger.mark("Create subscription");
+
+        final List<TestStreamingClient> clients = createSteamingClients(
+                subscription, Math.min(EVENT_TYPE_COUNT + 5, EVENT_TYPE_COUNT * PARTITION_PER_EVENT_TYPE));
```

Comment:
I was playing with different values for `EVENT_TYPE_COUNT` and `PARTITION_PER_EVENT_TYPE` and then stopped it. Now clients are not evenly distributed across event types, but they are somehow distributed between different event types and partitions. From the other hand it will be still limited by event types count as well